configuration/config2/EnvironmentMgr.cpp

```diff
@@ -0,0 +1,122 @@
+/*##############################################################################
+
+HPCC SYSTEMS software Copyright (C) 2015 HPCC Systems�.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+############################################################################## */
+
+#include "EnvironmentMgr.hpp"
+#include "ConfigExceptions.hpp"
+#include "XMLEnvironmentMgr.hpp"
+
+
+
+EnvironmentMgr *getEnvironmentMgrInstance(const std::string &envType, const std::string &configPath)
+{
+    EnvironmentMgr *pEnvMgr = NULL;
+    if (envType == "XML")
```

Comment:
can these types be enumerated? (I'm sure you currently only have xml, but let's set up for the future).